src/js/components/Calendar/Calendar.js

```diff
@@ -268,47 +257,94 @@ const Calendar = forwardRef(
           if (Array.isArray(d)) {
             let from;
             let to;
-            [from, to] = d.map((day) => new Date(day));
-            from = normalizeForTimezone(from, d[0]);
-            to = normalizeForTimezone(to, d[0]);
+            [from, to] = d.map((day) => day);
+            from = normalizeForTimezone(from, timestamp);
+            to = normalizeForTimezone(to, timestamp);
             dateArray.push([from, to]);
           } else {
-            dateArray.push(normalizeForTimezone(d));
+            dateArray.push(normalizeForTimezone(d, timestamp));
           }
         });
         return dateArray;
       }
       return undefined;
     };
 
+    let timestamp = timestampProp;
```

Comment:
Can we put this inside a `useMemo()`? This will avoid re-calculating as the user traverses around the Calendar without selecting anything.